packages/ra-core/src/reducer/admin/references/oneToMany.ts

```diff
@@ -114,6 +129,46 @@ export const getReferencesByIds = (
     return Object.keys(references).length > 0 ? references : null;
 };
 
+const getRelatedReferences = (previousState, resource) =>
+    Object.keys(previousState).filter(key => key.includes(resource));
+
+const removeDeletedReference = (previousState, action: ActionTypes) => {
+    const relatedTo = getRelatedReferences(previousState, action.meta.resource);
+
+    return relatedTo.reduce(
+        (acc, key) => ({
+            ...acc,
+            [key]: {
+                ids: previousState[key].ids.filter(
+                    id => id !== action.payload.id
+                ),
+                total: previousState[key].total - 1,
+            },
+        }),
+        previousState
+    );
+};
+
+const removeDeletedReferences = (previousState, action: ActionTypes) => {
+    const relatedTo = getRelatedReferences(previousState, action.meta.resource);
+
+    return relatedTo.reduce((acc, key) => {
+        const idsToRemove = previousState[key].ids.filter(id =>
+            action.payload.ids.includes(id)
+        );
+
+        return {
+            ...acc,
+            [key]: {
+                ids: previousState[key].ids.filter(
+                    id => !action.payload.ids.includes(id)
```

Comment:
haven't you just done that a couple lines earlier?

Reply:
The inverse